src/erlcloud_as.erl

```diff
@@ -367,6 +373,44 @@ resume_processes(GroupName, ScalingProcesses, Config) ->
     end.
 
 
+%% --------------------------------------------------------------------
+%% @doc Detach the given instances from the group using the default configuration
+%% without decrementing the desired capacity of the group.
+%% @end
+%% --------------------------------------------------------------------
+-spec detach_instances(list(string()),string()) -> aws_autoscaling_activity().
+detach_instances(InstanceIds, GroupName) ->
+    detach_instances(InstanceIds, GroupName, erlcloud_aws:default_config()).
+
+%% --------------------------------------------------------------------
+%% @doc Detach the given instances from the group.  The 3rd parameter can be:
+%% 'false' to not decrement the desired capacity of the group
+%% 'true' to decrement the desired capacity of the group
+%% Config a supplied AWS configuration.
+%% @end
+%% --------------------------------------------------------------------
+detach_instances(InstanceIds, GroupName, false) ->
```

Comment:
Actually I don't like the set of functions below. What about replacing them with just one with guard expression, i.e.
```
detach_instances(InstanceIds, GroupName, ShouldDecrementDesiredCapacity) when is_bool(ShouldDecrementDesiredCapacity) ->
    detach_instances(InstanceIds, GroupName, ShouldDecrementDesiredCapacity, erlcloud_aws:default_config())
```